packages/loader/driver-utils/src/buildSnapshotTree.ts

```diff
@@ -33,7 +32,7 @@ function flattenCore(
 			blobMap.set(id, buffer);
 
 			const entry: IGitTreeEntry = {
-				mode: FileMode[treeEntry.mode],
+				mode: treeEntry.mode,
```

Comment:
This seems correct since `treeEntry.mode` is already of type `FileMode`, I guess I'm just surprised the compiler didn't catch this before? I might be doing something wrong but [in TS Playground](https://www.typescriptlang.org/play/?#code/KYOwrgtgBBCeCi5oG8oFgBQUCCUC8UA5AIyEA0mUAQvkQEyGYC+mmAliAC7ABOAZgEMAxsCgB7KMkwBIAA48xs4gC4YCJM1YYhYkAGdO4leNqp5i43ESQAdLhYZMO-WIA2wG67EBzABRjiG3MlAEoAbiddPTcPLz8rJABtAKCFJQBdcMwgA) I can't get something like the earlier version to even compile, and the runtime value is different in each case. Leaves me a bit concerned about what this code was actually doing and if this is a behavioral change.

![image](https://github.com/user-attachments/assets/0dfca955-b39f-4661-bc75-6f2e754887d9)